### PR TITLE
WIP: migrate CI from TravisCI to GitHub Actions

### PR DESCRIPTION
**Pull request by jelhan**
GitHub Action workflow is autogenerated by [create-github-actions-setup-for-ember-addon](https://github.com/jelhan/create-github-actions-setup-for-ember-addon) using `v0.2.0`.

Keeping it in WIP for now because awaiting review of the generated GitHub Actions workflow.

### CODE REVIEW

**Comment by jelhan** [comment]
@onechiporenko Are you considering switching from TravisCI to GitHub Actions? If so I'm happy to use this repository as a testing for the script I have written.

GitHub Actions do not run automatically for pull requests created from forks even if configured to do so. It must be enabled explicitly in repository settings:

> Workflows don't run on forked repositories by default. You must enable GitHub Actions in the Actions tab of the forked repository.

See [GitHub Actions docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories) for more details.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-
+    - name: Install Dependencies
+      run: npm ci
+      if: |
+        steps.cache-global-package-manager-cache.outputs.cache-hit != 'true' ||
+        steps.cache-node-modules.outputs.cache-hit != 'true'
```
Comment (ijlee2) [inline .github/workflows/ci.yml:43]: Suggestions

- Use `actions/cache@v2` to simplify steps
- For `npm`, be careful with caching if the OS includes Windows, not just Ubuntu/Mac

https://gist.github.com/ijlee2/fba20304670b8d2b71f0b81cd07ea026#file-ci-yml-L71-L83

Comment (ijlee2) [reply .github/workflows/ci.yml:43]: The key (line 28) includes the word `yarn`. In the updated workflow template (linked above), I removed the word `yarn`/`npm` for convenience.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    - name: Lint
+      run: npm run-script lint
+
+  test:
```
Comment (ijlee2) [inline .github/workflows/ci.yml:47]: Suggestions for `test`:

- Decide whether (a) `test` runs only when `lint` succeeds, or (b) `test` can run in parallel with `lint`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+          ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-
+    - name: Install Dependencies
+      run: npm ci
+      if: |
+        steps.cache-global-package-manager-cache.outputs.cache-hit != 'true' ||
+        steps.cache-node-modules.outputs.cache-hit != 'true'
+    - name: Test
+      run: npm run-script test:ember --launch ${{ matrix.browser }}
+
+  floating-dependencies:
```
Comment (ijlee2) [inline .github/workflows/ci.yml:91]: Suggestions for `floating-dependencies`:

- Remove caching and using `package-lock.json` to install dependencies

**Comment by ijlee2** [comment]
@jelhan It looked like most of the workflow approach was the same as in https://github.com/jelhan/ember-style-modifier/pull/32, the only difference being `yarn` vs `npm`. I think it'll be okay if you refer to the other pull request for more detailed feedback.

**Comment by onechiporenko** [comment]
Hello. Yes, I'm ready to switch to Github-actions. And I appreciate you help me with this. Thanks!

**Comment by jelhan** [comment]
> Hello. Yes, I'm ready to switch to Github-actions. And I appreciate you help me with this. Thanks!

Great to hear! Would be awesome if you can active them for pull requests created by forks in the repository settings. This will make testing for me easier. :smile:

**Comment by onechiporenko** [comment]
@jelhan. It's enabled.

Something like `on: [push, pull_request]` should be in config in the PR.

**Pull request closed without merging**